x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeRequest.java

```diff
@@ -40,6 +40,17 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeBoolean(sourcesFinished);
     }
 
+    @Override
+    public TaskId getParentTask() {
+        // Exchange requests with `sourcesFinished=true` complete the remote sink and return without blocking.
+        // Masking the parent task allows these requests to bypass task cancellation, ensuring cleanup of the remote sink.
+        // TODO: Maybe add a separate action/request for closing exchange sinks?
```

Comment:
I wonder indeed if we shouldn't drop `sourcesFinished` and instead have ExchangeCloseRequest or something?

Reply:
Yes, my plan is to remove sourcesFinished and replace it with ExchangeCloseRequest.